### DISCUSSION TITLE
Assets-request: allow format in combination with presets that use format=auto

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#19470] [Feature Requests]
### Summary

Currently it's not possible to combine any transformation preset with custom transformation parameters. In case a preset contains format=auto I think it would make sense to allow the enduser to specify the wished formt manually.

### Basic Example

In some cases we do want a specific format. In those it would be great if we could still use our preset and just specify the format.

### Motivation

More flexibility with file formats

### Detailed Design

Override the format by passing a format key. Optionally it could even be possible to combine every transformation key with presets by overriding the specific parts.

### Must Have:
- Override format in case format=auto

### Could Have:
- Option to override all transformation keys

### Drawbacks

/

### Alternatives

Request the file-format via the SEO-Friendly name

### Adoption Strategy

/

**Comment by rijkvanzanten** [comment]
I think it makes a ton of sense to me that you should be able to override any of the presetted fields by adding more query params, like `key=thumbnail&width=500` (use thumbnail but at 500 width). (Assuming you have the project settings configured to allow any presets and not just keys) 🙂

**Comment by rijkvanzanten** [comment]
Gonna mark this as a small improvement to the assets endpoint rather than a full new feature, as it's tiny in scope and a known change. We'll track the implementation of this in https://github.com/directus/directus/issues/19485

**Reply by Dominic-Marcelino** [reply]
Happy to read this! 🥳
